dandi/dandiapi.py

```diff
@@ -269,21 +287,27 @@ def dandi_authenticate(self):
                 break
 
     def get_dandiset(
-        self, dandiset_id: str, version_id: Optional[str] = None
+        self, dandiset_id: str, version_id: Optional[str] = None, lazy: bool = True
     ) -> "RemoteDandiset":
         """
         Fetches the Dandiset with the given ``dandiset_id``.  If ``version_id``
         is not specified, the `RemoteDandiset`'s version is set to the most
         recent published version if there is one, otherwise to the draft
         version.
+
+        If ``lazy`` is true, no requests are actually made until any data is
+        requested from the `RemoteDandiset`.
         """
-        d = RemoteDandiset._make(self, self.get(f"/dandisets/{dandiset_id}/"))
-        if version_id is not None and version_id != d.version_id:
-            if version_id == "draft":
-                d.version = d.draft_version
-            else:
-                return d.for_version(version_id)
-        return d
+        if lazy:
+            return RemoteDandiset(self, dandiset_id, version_id)
+        else:
+            d = RemoteDandiset._make(self, self.get(f"/dandisets/{dandiset_id}/"))
+            if version_id is not None and version_id != d.version_id:
+                if version_id == DRAFT:
+                    d.version = d.draft_version
+                else:
+                    return d.for_version(version_id)
+            return d
```

Comment:
coverage is odd here but suggests that we do not cover non-lazy code block.  Please add a dedicated test

Reply:
Done.